packages/edit-site/src/components/global-styles/test/utils.js

```diff
@@ -134,13 +134,73 @@ describe( 'editor utils', () => {
 		describe( 'when provided a custom variable', () => {
 			it( 'retrieves the correct custom value', () => {
 				const actual = getValueFromVariable(
-					styles,
+					themeJson,
 					'root',
 					'var(--wp--custom--color--secondary)'
 				);
 
 				expect( actual ).toBe( '#a65555' );
 			} );
 		} );
+
+		describe( 'when provided a dynamic reference', () => {
+			it( 'retrieves the referenced value', () => {
+				const stylesWithRefs = {
+					...themeJson,
+					styles: {
+						color: {
+							background: {
+								ref: 'styles.color.text',
+							},
+							text: 'purple-rain',
+						},
+					},
+				};
+				const actual = getValueFromVariable( stylesWithRefs, 'root', {
+					ref: 'styles.color.text',
+				} );
+
+				expect( actual ).toBe( stylesWithRefs.styles.color.text );
+			} );
+
+			it( 'returns where value is dynamic reference and reference does not exist', () => {
```

Comment:
should this be 'returns the originally provided value where value is ....?

Reply:
yes it should. Thanks!